crates/chia_streamable_macro/src/lib.rs

```diff
@@ -1,15 +1,127 @@
 extern crate proc_macro;
 
+use proc_macro::TokenStream;
 use proc_macro2::{Ident, Span};
 use proc_macro_crate::{crate_name, FoundCrate};
 use quote::quote;
+use syn::token::Pub;
 use syn::Lit::Int;
 use syn::{
     parse_macro_input, Data, DeriveInput, Expr, Fields, FieldsNamed, FieldsUnnamed, Index, Type,
+    Visibility,
 };
 
+#[proc_macro_attribute]
+pub fn streamable(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let found_crate =
+        crate_name("chia-protocol").expect("chia-protocol is present in `Cargo.toml`");
+
+    let chia_protocol = match found_crate {
+        FoundCrate::Itself => quote!(crate),
+        FoundCrate::Name(name) => {
+            let ident = Ident::new(&name, Span::call_site());
+            quote!(#ident)
+        }
+    };
+
+    let is_message = &attr.to_string() == "message";
+
+    let my_name = std::env::var("CARGO_PKG_NAME").unwrap();
+    let is_protocol = &my_name == "chia-protocol";
```

Comment:
this looks like a slightly different way to do the same thing as `match found_crate` above. it looks redundant.

Reply:
Good point, I'll consolidate it